aws-http-nio-client-netty/src/main/java/software/amazon/awssdk/http/nio/netty/EventLoopGroupBuilder.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.nio.netty;
+
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.epoll.Epoll;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import java.util.concurrent.ThreadFactory;
+import software.amazon.awssdk.utils.builder.SdkBuilder;
+
+/**
+ * Builder for a Netty {@link EventLoopGroup}. Will use the optimal {@link EventLoopGroup} for the
+ * given platform.
+ */
+public final class EventLoopGroupBuilder {
+
+    /**
+     * Create a {@link Builder} that can be used to create an immutable {@link EventLoopGroupBuilder}.
+     */
+    public static Builder builder() {
+        return new DefaultBuilder();
+    }
+
+    /**
+     * Builder interface for {@link EventLoopGroupBuilder}.
+     */
+    public interface Builder extends SdkBuilder<Builder, EventLoopGroup> {
+
+        /**
+         * Number of threads to use for the {@link io.netty.channel.EventLoopGroup}. If not set, the default
+         * thread count is two times the number of processors available.
```

Comment:
Extreme nit: "two times" sounds weird, I'd have gone with "double the number" or "twice the number"

Also this is really just the netty default right? Looking at the netty code it can also come from a system property - https://github.com/netty/netty/blob/4.1/transport/src/main/java/io/netty/channel/MultithreadEventLoopGroup.java#L40, falling back to double the processors.

Maybe we just say "If not set, the Netty default thread count is used"